The Learning Servlet

omniprof opened the pull request:
Corrected minor error in pom.xml file.
Added a new example to the Servlet Examples module called TheLearningServlet that is designed for beginners to servlets. 
Signed-off-by:Ken Fogel <omniprof@gmail.com>

manorrock commented on servlet/theLearningServlet/pom.xml:
```diff
@@ -0,0 +1,86 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Please remove the comments from the POM.

If you want to example Maven I would suggest linking to the Maven book by Sonatype.

manorrock commented on servlet/theLearningServlet/src/test/java/jakartaee/examples/servlet/thelearningservlet/TheLearningServletTest.java:
```diff
@@ -0,0 +1,108 @@
+/*
```
Please do not use @ in JavaDoc comments unless linking to documentation that explains something

manorrock commented on servlet/pom.xml:
```diff
@@ -19,6 +19,7 @@
         <module>pushBuilder</module>
         <module>servletContextListener</module>
         <module>servletRequestListener</module>
+        <module>theLearningServlet</module>
```
Please rename theLearningServlet to explainingHttpServlet (as that is what you are explaining here)

omniprof commented:
I have made all the requested changes. You mention the license header and I believe, based on reviewing other examples, it only appears in the Java source files and not the html, css, pom.xml or web.xml files. The new name does fit the style of the other examples.

I do object to the request to remove comments from the pom file and the html file although I have complied. If the audience to these examples are experienced developers looking to see what's new in Jakarta then maybe you can make the case that commenting these files is superfluous. I contend that many of the users of these examples are students and new developers whom we have a duty to provide with as much information about what is happening in these examples as possible.

manorrock commented on servlet/explainingHttpServlet/src/main/java/jakartaee/examples/servlet/explainingHttpServlet/ExplainingHttpServlet.java:
```diff
@@ -10,14 +10,14 @@
+ * @author Ken Fogel (omniprof@gmail.com)
  *
- * Twitter: @omniprof - Blog: www.omnijava.com
+ * Twitter: omniprof - Blog: www.omnijava.com
```
Sorry I missed this one ... none of the examples add more than the @author tag so can you remove this and then I'll approve the PR and I'll ask @arjantijms Arjan to merge it.

manorrock commented:
Ken your objection is noted and I do understand it, but these examples are not a guide on learning Maven, HTML or CSS. If you think we should list links to each of those respective topics in our main README I am OK with that.

pzygielo commented on servlet/theLearningServlet/src/main/java/jakartaee/examples/servlet/thelearningservlet/TheLearningServlet.java:
```diff
@@ -0,0 +1,270 @@
+    private static final Logger LOG = Logger.getLogger(TheLearningServlet.class.getName());
+
+    /**
+     * Constructor
```
I don't think this is needed. Not only it is logged in the body, but javadoc prepares separate section for constructors.

pzygielo commented on servlet/theLearningServlet/src/main/java/jakartaee/examples/servlet/thelearningservlet/TheLearningServlet.java:
```diff
@@ -0,0 +1,270 @@
+     * Constructor
+     *
+     * Can only be used to initialize class variables or perform tasks before
+     * the servlet comes into existence. Class variables are not thread safe so
```
It doesn't say so, but for me it suggests that in opposition - instance variables are thread safe. But are they?

pzygielo commented on servlet/theLearningServlet/src/main/java/jakartaee/examples/servlet/thelearningservlet/TheLearningServlet.java:
```diff
@@ -0,0 +1,270 @@
+     * copyright that you can add. Usually called by the app server similar to
+     * how the attribute 'description' is used in @WebServlet.
+     *
+     * @return
```
I'd drop empty javadoc tags, as they provide no additional info, like this return above, or params/throws below.

omniprof commented:
All items brought to my attention by pzygielo are addressed.

omniprof commented:
I have addressed all the items you brought to my attention.

 

Thank-you,

 

Ken

 

From: Piotrek Żygieło <notifications@github.com> 
Sent: May 3, 2020 4:03 PM
To: eclipse-ee4j/jakartaee-examples <jakartaee-examples@noreply.github.com>
Cc: Ken Fogel <omniprof@gmail.com>; Author <author@noreply.github.com>
Subject: Re: [eclipse-ee4j/jakartaee-examples] The Learning Servlet (#20)

 

@pzygielo commented on this pull request.

  _____  

In servlet/theLearningServlet/src/main/java/jakartaee/examples/servlet/thelearningservlet/TheLearningServlet.java <https://github.com/eclipse-ee4j/jakartaee-examples/pull/20#discussion_r419152470> :

> +     * were available to all threads of the servlet that you likely created in
+     * init().
+     */
+    @Override
+    public void destroy() {
+        LOG.info(">>> destroy <<<");
+    }
+
+    /**
+     * getter for a String of information
+     *
+     * Returns information about the servlet, such as author, version, and
+     * copyright that you can add. Usually called by the app server similar to
+     * how the attribute 'description' is used in @WebServlet.
+     *
+     * @return

I'd drop empty javadoc tags, as they provide no additional info, like this return above, or params/throws below.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/eclipse-ee4j/jakartaee-examples/pull/20#pullrequestreview-404652482> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AA7AMUM34FPRXO6RPYWJ6BLRPXEVPANCNFSM4MVU7FDQ> .  <https://github.com/notifications/beacon/AA7AMUJZFSCRP3UHKBU3EN3RPXEVPA5CNFSM4MVU7FD2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGODAPIDQQ.gif>

pzygielo commented:
> I have addressed all the items you brought to my attention. 

1. I can't see the value of empty javadoc tags, that do not provide explanations.
Unnecessary to duplicate param names or throw declarations.
Empty return tag?

It all results in:
```
...: warning: no description for @param
...: warning: no description for @throws
...: warning: no description for @return
```
from `javadoc`.

2. There is no need for FQDN in javadoc, types get resolved properly from imports.

pzygielo commented:
> It all results in:
> ...

With current project configuration (i.e. implicit javadoc plugin 3.2.0), `mvn javadoc:javadoc` results in build error.

pzygielo commented on servlet/explainingHttpServlet/src/main/java/jakartaee/examples/servlet/explainingHttpServlet/ExplainingHttpServlet.java:
```diff
@@ -0,0 +1,267 @@
+     */
+    /**
+     * Called by the server (via the service method) to allow a servlet to
+     * handle a GET request. In a RESTful web service it is CRUD -> R
```
javadoc doesn't like this:
```
jakartaee-examples/servlet/explainingHttpServlet/src/main/java/jakartaee/examples/servlet/explainingHttpServlet/ExplainingHttpServlet.java:171: error: bad use of '>'
     * handle a POST request. In a RESTful web service it is CRUD -> C
                                                                   ^
```

omniprof commented:
Removed '->' from comments
Added info to @return in getServletInfo()
Added info to four doXXX methods describing where request and response come from.
JavaDocs shows a warning for not describing what the exceptions are for. Leaving this unchanged.

Will add creating JavaDocs to my blog and submitting to an open source project.

arjantijms commented:
That was quite a process, but let's merge it in now! :)

omniprof commented:
It was worth all the time I spent. Thank you to everyone who reviewed the code and went to the trouble of finding ways I could make it better.

________________________________
From: Arjan Tijms <notifications@github.com>
Sent: Tuesday, May 5, 2020 5:29:10 PM
To: eclipse-ee4j/jakartaee-examples <jakartaee-examples@noreply.github.com>
Cc: Ken Fogel <omniprof@gmail.com>; Author <author@noreply.github.com>
Subject: Re: [eclipse-ee4j/jakartaee-examples] The Learning Servlet (#20)


Merged #20<https://github.com/eclipse-ee4j/jakartaee-examples/pull/20> into master.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/eclipse-ee4j/jakartaee-examples/pull/20#event-3306088212>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AA7AMUKNKRZV26FIQNYPJHTRQCAKNANCNFSM4MVU7FDQ>.
